Bottom: 5502795e78c47867f0ae0974f6b563b634f224d1
Top:    c16520ee054a920f8409303746f773a9d3b23fb7
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 21:47:49 +0000

Refresh of adjust-coordinate-system

---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 08c511d..c843fab 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -24,6 +24,7 @@
 #define PERFECT_ROUND_CONTOURS
 
 #define REVERSED_PCB_CONTOURS 1 /* PCB Contours are reversed from the expected CCW for outer ordering - once the Y-coordinate flip is taken into account */
+#undef REVERSED_PCB_CONTOURS
 
 #ifdef REVERSED_PCB_CONTOURS
 #define COORD_TO_STEP_X(pcb, x) (COORD_TO_MM(                   (x)))
@@ -380,7 +381,7 @@ object3d_from_contours (const POLYAREA *contours,
 #ifdef REVERSED_PCB_CONTOURS
       face3d_set_normal (faces[npoints], 0., 0., -1.);
 #else
-      face3d_set_normal (faces[npoints], 0., 0., 1.);
+      face3d_set_normal (faces[npoints], 0., 0., 1.); /* PCB bottom is at positive Z in this scheme */
 #endif
       face3d_set_appearance (faces[npoints], top_bot_appearance);
       object3d_add_face (object, faces[npoints]);
@@ -389,7 +390,7 @@ object3d_from_contours (const POLYAREA *contours,
 #ifdef REVERSED_PCB_CONTOURS
       face3d_set_normal (faces[npoints + 1], 0., 0., 1.);
 #else
-      face3d_set_normal (faces[npoints + 1], 0., 0., -1.);
+      face3d_set_normal (faces[npoints + 1], 0., 0., -1.); /* PCB top is at negative Z in this scheme */
 #endif
       face3d_set_appearance (faces[npoints + 1], top_bot_appearance);
       object3d_add_face (object, faces[npoints + 1]);
diff --git a/src/hid/step/object3d_step.c b/src/hid/step/object3d_step.c
index 829221b..d47a8b4 100644
--- a/src/hid/step/object3d_step.c
+++ b/src/hid/step/object3d_step.c
@@ -26,6 +26,7 @@
 
 
 #define REVERSED_PCB_CONTOURS 1 /* PCB Contours are reversed from the expected CCW for outer ordering - once the Y-coordinate flip is taken into account */
+#undef REVERSED_PCB_CONTOURS
 
 #define EPSILON 1e-5 /* XXX: Unknown  what this needs to be */
